libbeat/processors/decompress/gzip.go

```diff
@@ -0,0 +1,147 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package gzip
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/beat"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/processors"
+	"github.com/elastic/beats/libbeat/processors/checks"
+
+	"bytes"
+	"compress/gzip"
+	"io"
+)
+
+type decompressGzipFields struct {
+	decompressGzipFieldsConfig
+	fields map[string]string
+}
+
+type decompressGzipFieldsConfig struct {
+	Fields        common.MapStr `config:"fields"`
+	IgnoreMissing bool          `config:"ignore_missing"`
+	OverwriteKeys bool          `config:"overwrite_keys"`
+	FailOnError   bool          `config:"fail_on_error"`
+}
+
+var (
+	defaultDecompressGzipFieldsConfig = decompressGzipFieldsConfig{
+		FailOnError: true,
+	}
+)
+
+func init() {
+	processors.RegisterPlugin("decompress_gzip_fields",
+		checks.ConfigChecked(NewDecompressGzipFields,
+			checks.RequireFields("fields"),
+			checks.AllowedFields("fields", "ignore_missing", "overwrite_keys", "overwrite_keys", "fail_on_error", "when")))
+}
+
+// NewDecompressGzipFields construct a new decompress_gzip_fields processor.
+func NewDecompressGzipFields(c *common.Config) (processors.Processor, error) {
+	config := defaultDecompressGzipFieldsConfig
+
+	err := c.Unpack(&config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unpack the decompress_gzip_fields configuration: %s", err)
```

Comment:
Please use `%+v` in format strings when reporting errors.

Reply:
Replaced